lib/trino-parquet/src/main/java/io/trino/parquet/reader/FilteredRowRanges.java

```diff
@@ -13,28 +13,115 @@
  */
 package io.trino.parquet.reader;
 
+import com.google.common.annotations.VisibleForTesting;
 import org.apache.parquet.internal.filter2.columnindex.RowRanges;
 
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 
 public class FilteredRowRanges
 {
     private final RowRanges parquetRowRanges;
+    private final List<RowRange> rowRanges;
```

Comment:
it seems this change belongs to previous commit or this class should be introduced in this commit

Reply:
This change needs to be made here as it's about using column index in batched reader.
Batched reader must adhere to row ranges from the index in order to be work correctly side by side with existing column readers.
I introduced the wrapper class in a previous commit so that it is easier to see which part is specifically introduced for batched reader